### PR TITLE
Update analysis-common module to use new rest test framework

### PR DESCRIPTION
**Pull request by brianseeders**
See title

### CODE REVIEW

**Comment by brianseeders** [comment]
run elasticsearch-ci/part-1

**Review by mark-vieira — APPROVED** [review]

File: modules/analysis-common/build.gradle [hunk 1]
```diff
@@ -23,13 +23,11 @@ restResources {
...
-testClusters.configureEach {
-  module ':modules:reindex'
-  module ':modules:mapper-extras'
-}
-
 dependencies {
   compileOnly project(':modules:lang-painless:spi')
+  clusterModules project(':modules:reindex')
+  clusterModules project(':modules:mapper-extras')
+  clusterModules project(':modules:analysis-common')
```
[the remark this answers is not in the document]
Comment (mark-vieira) [reply modules/analysis-common/build.gradle:30]: Yeah, we should think about eliminating the need for that as well. Perhaps passing in system property to the test that indicates "default" plugins/modules that should be installed.

**Comment by brianseeders** [comment]
@elasticmachine update branch

**Comment by brianseeders** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.11|[](https://github.com/elastic/elasticsearch/pull/101541)|
|✅|8.10|[](https://github.com/elastic/elasticsearch/pull/101542)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**